viper/function_signature.py

```diff
@@ -70,16 +70,23 @@ def from_definition(cls, code):
                 pos += get_size_of_type(parsed_type) * 32
 
         # Apply decorators
-        const, payable, internal = False, False, False
+        const, payable, internal, public = False, False, False, False
         for dec in code.decorator_list:
             if isinstance(dec, ast.Name) and dec.id == "constant":
                 const = True
             elif isinstance(dec, ast.Name) and dec.id == "payable":
                 payable = True
             elif isinstance(dec, ast.Name) and dec.id == "internal":
                 internal = True
+            elif isinstance(dec, ast.Name) and dec.id == "public":
+                public = True
             else:
                 raise StructureException("Bad decorator", dec)
+        if public and internal:
+            raise StructureException("Cannot use public and internal decorators on the same function")
+        if not public and not internal and not isinstance(code.body[0], ast.Pass):
+            # import pdb; pdb.set_trace()
+            raise StructureException("Function visibility must be declared")
```

Comment:
So, this will throw if neither @public nor @internal is set? Am I reading this right?

I thought the idea was to remove internal by defaulting to public?

Reply:
I just worry we end up with too many decorators, which ends up looking like solidity's modifiers (although it can't be user-defined for viper).

Reply:
@DavidKnott I also see a stuck pdb there, please remove :stuck_out_tongue: